### PR TITLE
Add doc about the pip-to-conda mapping that can be provided in setup.cfg

### PR DESCRIPTION
**Pull request by maximlt**

### CODE REVIEW

File: instructions.md [hunk 1]
```diff
@@ -102,6 +102,17 @@ project_urls =
...
+in the `setup.cfg` file as follows:
+
+```
+[tool:pyctdev.conda]
+namespace_map = 
+    pip_package1=conda_package1
+    pip_package2=conda_package2
+    ...
+```
+
```
Comment (jbednar) [inline instructions.md:115]: Thanks! Can you add an example here with some concrete package that differs between the ecosystems? Usually it's "something" vs "something-base", or "pysomething" vs "something"...

Comment (maximlt) [reply instructions.md:115]: Replaced the dummy examples by two concrete examples.

Comment (jbednar) [reply instructions.md:115]: How does it work for a pip package with optional extras like `pkg[recommended]`?

Comment (maximlt) [reply instructions.md:115]: Just tried by declaring this in the `setup.py` of hvplot:
```python
extras_require['dummy'] = ['ibis[sqlite]']
```
and this in setup.cfg:
```python
[tool:pyctdev.conda]
namespace_map = 
    ibis[sqlite]=ibis-sqlite
```
and running `doit develop_install -c conda-forge -o dummy`.

It failed with:
```
PackagesNotFoundError: The following packages are not available from current channels:

  - ibis

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.


TaskFailed - taskid:develop_install
Command failed: 'conda install -y  -c conda-forge "bokeh >=1.0.0" "colorcet >=2" "holoviews >=1.11.0" "pandas" "numpy >=1.15" "ibis"' returned 1
```

So the extra in bracket is stripped by pyctdev :/ Some explanation here I think: https://github.com/pyviz-dev/pyctdev/blob/61d0e3320628e2e5228422321ffb2ccb824ae446/pyctdev/_conda.py#L397-L412

Comment (maximlt) [reply instructions.md:115]: Added a note about this.

**Pull request merged**